library/modules/Items.cpp

```diff
@@ -681,6 +687,81 @@ static void addQuality(std::string &tmp, int quality)
     }
 }
 
+//  It's not impossible the functionality of this operation is provided by one of the unmapped item functions.
+std::string Items::getTitle(df::item *item)
+{
+    CHECK_NULL_POINTER(item);
+
+    std::string tmp;
+
+    if (item->getType() == df::item_type::BOOK)
+    {
+        if (virtual_cast<df::item_bookst>(item)->title != "")
```

Comment:
`virtual_cast` is a non-trivial function, so you should call it once and store the result in a local variable.

Reply:
you can also do `if (auto book = virtual_cast<df::item_bookst>(item))` as the first condition and then just use `book`